pennylane/qnode.py

```diff
@@ -910,9 +937,15 @@ def to_autograd(self):
 
         if self.interface != "autograd" and self.interface is not None:
             # Since the interface is changing, need to re-validate the tape class.
-            self._tape, interface, self.device, diff_options = self.get_tape(
-                self._original_device, "autograd", self.diff_method
-            )
+            # if method was changed from "best", set it back to best
+            if self.diff_method_change:
+                self._tape, interface, self.device, diff_options = self.get_tape(
+                    self._original_device, "autograd", "best"
+                )
+            else:
+                self._tape, interface, self.device, diff_options = self.get_tape(
+                    self._original_device, "autograd", self.diff_method
+                )

```

Comment:
Your suggestion was the original code. It will not work with the new function of diff_method, which is always to show the selected method instead of "best". Suppose we set it always to be "best" here. In that case, we will lose what the user wanted (if the user mentioned a specific method), and if we don't check for a change, the method which was selected (internally from "best") before the node being transformed into jax/torch/autograd will pass. Still, we do not want that, as for the re-validation of tape for these interfaces, a new method needs to be selected.

Reply:
I just meant changing it without changing any logic, like this:
```suggestion
            # if method was changed from "best", set it back to best
            if self.diff_method_change:
                diff_method = "best"
            else:
                diff_method = self.diff_method
            self._tape, interface, self.device, diff_options = self.get_tape(
                self._original_device, "autograd", diff_method
            )
```
This shouldn't change any logic at all, but to me it's slightly clearer. But again, this is a minor suggestion, and not too important.